src/dbus/methods/player_peer_info.c

```diff
@@ -0,0 +1,109 @@
+/**
+ * WarfaceBot, a blind XMPP client for Warface (FPS)
+ * Copyright (C) 2015-2017 Levak Borok <levak92@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <gio/gio.h>
+#include <wb_dbus_methods.h>
+#include <wb_xmpp_wf.h>
+#include <wb_tools.h>
+#include <wb_xmpp.h>
+
+
+struct cb_args
+{
+    Warfacebot *object;
+    GDBusMethodInvocation *invocation;
+};
+
+void cmd_peer_player_info_dbus_cb(const char *result,
+                                  void *args)
+{
+    struct cb_args *a = (struct cb_args *) args;
+
+    if (result == NULL)
+    {
+      result = "-1";
+    }
+
+    warfacebot_complete_player_peer_info(
+        a->object,
+        a->invocation,
+        result);
```

Comment:
I'm not proud of how I ended up with `BuddyWhois` (tuple of values), and I wonder if something better can be done instead of just returning a string with the full answer from the server. I don't know if DBus supports key-value pair / dictionary. I'm reading that `a{ss}` is a string-string dictionary, and `a{?*}` is a casual dictionary.
Ideally the parsing would have to be done by `xmpp_iq_profile_info_get_status` by giving the callback a structure (to create) with all the supported fields. Then the dbus handler would pack those values back into a variant.

https://developer.gnome.org/glib/stable/glib-GVariant.html#GVariantDict

It's just an idea.